### ISSUE TITLE
Anti-spam measures

### ISSUE THREAD

**Issue by Jawshy** [ArmaLife/Framework#746]
Certain actions can be executed many times per second by a player which can degrade performance and flood logs with messages. 

I suggest that certain actions be restricted to a certain number of uses per second to mitigate this issue. 

The following actions can be spammed and what gets spammed: 
- [Locking and unlocking vehicles](https://github.com/ArmaLife/Framework/blob/109ab407479f7ea3ab5463d1682ff3a5f7aa3978/Altis_Life.Altis/core/functions/fn_keyHandler.sqf#L232). 
  - systemChat
  - remoteExec (life_fnc_say3D)
- [Surrendering](https://github.com/ArmaLife/Framework/blob/109ab407479f7ea3ab5463d1682ff3a5f7aa3978/Altis_Life.Altis/core/functions/fn_keyHandler.sqf#L68)
  - setVariable (playerSurrender) 
- [3D preview inside vehicle shop](https://github.com/ArmaLife/Framework/blob/770a92e725fb404099ed342bf3f859c12c985371/Altis_Life.Altis/Config_Master.hpp#L117)
  - attachTo (vehicle class + Land_HelipadEmpty_F)
  - createVehicle (vehicle class)
  - remoteExec (life_fnc_hideobj)
  - setVariable (life_veh_color)

**Comment by BoGuu** [comment]
@Jawshy 

More info the second? Do you mean when they open the dialog?

**Comment by Jawshy** [comment]
@BoGuu, yes. When the player opens the shop and scrolls through the available vehicles and skins/textures.

**Comment by BoGuu** [comment]
Okay, well, the first two I'll whip up shortly. The 3D preview needs to be moved over to createVehicleLocal, god knows why it isn't already, this will remove any performance impact for other clients, and will make hideObj not needed. We can't stop them scrolling through, it's a shop after all.

**Comment by NiiRoZz** [comment]
Can't move to createvehiclelocal because create spam but Will mode to hideobjectglobal executed server side

**Comment by BoGuu** [comment]
@NiiRoZz 

I'm not quite understand what you are saying. Why would createVehicleLocal create spam? It would stop all un-needed broadcasting, and would remove the need to hide the object for others?

**Comment by NiiRoZz** [comment]
When put createvehicleLocal inside rpt server you have spam see #357

**Comment by BoGuu** [comment]
As @Jawshy pointed out, what you've done has just escalated the problem. You're broadcasting more over the network, and involving the remoteExec logs?

**Comment by NiiRoZz** [comment]
I Removed the spam inside all rpt, i just don't Remove the battleye spam but i Will do

**Comment by BoGuu** [comment]
It's some spam in the client log, it really doesn't matter, BI's fault, happens, but reducing server performance and causing more spam in the server files as a result, doesn't seem logical?
